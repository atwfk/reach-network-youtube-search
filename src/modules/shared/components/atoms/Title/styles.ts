import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledTitle = styled.h3`
  font-weight: 400;
  font-size: 16px;

  @media (min-width: ${SCREENS.MD}) {
    font-size: 25px;
  }
`;
