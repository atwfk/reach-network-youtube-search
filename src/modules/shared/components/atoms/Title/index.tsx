import React from "react";
import type { FC, ReactElement } from "react";
import { StyledTitle } from "./styles";

const Title: FC<{ title: string }> = ({ title }): ReactElement => {
  return (
    <StyledTitle className="title" title={title}>
      {title}
    </StyledTitle>
  );
};

export default Title;
