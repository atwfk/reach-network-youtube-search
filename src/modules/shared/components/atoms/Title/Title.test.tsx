import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Title from ".";

afterEach(cleanup);

describe("Title Component snapshots", () => {
  it("Should render simple Title component", () => {
    const tree = renderer
      .create(<Title title="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
