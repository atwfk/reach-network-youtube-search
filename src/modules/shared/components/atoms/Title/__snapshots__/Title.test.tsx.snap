// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component snapshots Should render simple Title component 1`] = `
.c0 {
  font-weight: 400;
  font-size: 16px;
}

@media (min-width:768px) {
  .c0 {
    font-size: 25px;
  }
}

<h3
  className="c0 title"
  title="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
>
  Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]
</h3>
`;
